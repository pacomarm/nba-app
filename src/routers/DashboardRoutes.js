import React from 'react'
import {
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import {NavBar} from '../components/ui/NavBar'
import { LAScreen } from '../components/lakers/LAScreen';
import { PlayerScreen } from '../components/players/PlayerScreen';
import { BullsScreen } from '../components/bulls/BullsScreen';

export const DashboardRoutes = () => {
    return (
        <>
          <NavBar/>
          <div className="ml-5 mt-5">
              <Switch>
                  <Route exact path="/lakers" component={LAScreen}></Route>
                  <Route exact path="/player/:playerId" component={PlayerScreen}></Route>
                  <Route exact path="/bulls" component={BullsScreen}></Route>
                  <Redirect to="/lakers"/>
              </Switch>
          </div>
        </>
    )
}
