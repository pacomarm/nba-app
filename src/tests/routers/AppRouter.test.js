import React from 'react';
import { mount, configure } from "enzyme";
import { AppRouter } from '../../routers/AppRouter';
import { MemoryRouter } from 'react-router-dom';

import Adapter, { prototype } from '@wojtekmaj/enzyme-adapter-react-17';
import { AuthContext } from '../../auth/AuthContext';
configure({adapter: new Adapter()});

describe('tests AppRouter', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('should show login if not Auth', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should show Lakers component if Auth', () => { 

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Paco'
            }
        }
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe(true);
    })

    test('should ', () => {
    
    })
    
})
