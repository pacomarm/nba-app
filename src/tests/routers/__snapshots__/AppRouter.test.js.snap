// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests AppRouter should show login if not Auth 1`] = `ReactWrapper {}`;
